tests/manage/z_cluster/test_osd_heap_profile.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import pytest
+import time
+
+from ocs_ci.framework.testlib import ManageTest, tier1, skipif_ocs_version
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod, get_osd_pods, get_osd_pod_id
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_ocs_version("<4.7")
+@pytest.mark.polarion_id("OCS-2512")
+class TestOSDHeapProfile(ManageTest):
+    """
+    Test osd heap profile created on '/var/log/ceph/'.
+
+    """
+
+    @pytest.fixture(scope="function", autouse=True)
+    def teardown(self, request):
+        """
+        Archive a crash report so that it is no longer considered for the
+        RECENT_CRASH health check and does not appear in the
+        crash ls-new output
+
+        """
+
+        def finalizer():
+            logging.info("Silence the ceph warnings by “archiving” the crash")
+            tool_pod = get_ceph_tools_pod()
+            tool_pod.exec_ceph_cmd(ceph_cmd="ceph crash archive-all", format=None)
+            logging.info(
+                "Perform Ceph and cluster health checks after silencing the ceph warnings"
+            )
+            ceph_health_check()
+
+        request.addfinalizer(finalizer)
+
+    def test_osd_heap_profile(self):
+        """
+        Test osd heap profile created on '/var/log/ceph/'.
+
+        """
+        log.info("Start heap profiler for osd-0")
```

Comment:
can we pick a random osd instead of hardcoding it?

Reply:
done

Reply:
done